include/alp/algorithms/householder_tridiag.hpp

```diff
@@ -73,82 +103,125 @@ namespace alp {
 				// alpha = norm( v ) * v[ 0 ] / norm( v[ 0 ] )
 				// v = v - alpha * e1
 				// v = v / norm ( v )
-				Vector< D, structures::General, Dense > v;
-				rc = set( v, get_view( RR, utils::range( k + 1, n ), k ) );
 
-				Scalar< D > alpha;
+				auto v_view = get_view( RR, k, utils::range( k + 1, n ) );
+				Vector< D, structures::General, Dense > v( n - ( k + 1 ) );
+				rc = set( v, v_view );
+				if( rc != SUCCESS ) {
+					std::cerr << " set( v, view ) failed\n";
+					return rc;
+				}
+
+				Scalar< D > alpha( zero );
 				rc = norm2( alpha, v, ring );
+				if( rc != SUCCESS ) {
+					std::cerr << " norm2( alpha, v, ring ) failed\n";
+					return rc;
+				}
 
 				rc = eWiseLambda(
-							[ &v, &alpha ]( const size_t i ) {
-								if ( i == 0 ) {
-									Scalar< D > norm_v0( std::abs( v[ i ] ) );
-									foldl(alpha, v [ i ], ring.getMultiplicativeOperator() );
-									foldl(alpha, norm_v0, divide );
-									foldl(v [ i ], alpha, minus );
-								}
-							},
-							v );
-
-				Scalar< D > norm_v;
+					[ &alpha, &ring, &divide, &minus ]( const size_t i, D &val ) {
+						if ( i == 0 ) {
+							Scalar< D > norm_v0( std::abs( val ) );
+							internal::foldl( *alpha, val, ring.getMultiplicativeOperator() );
```

Comment:
Could we use `alp::foldl(alpha, ...)` instead of `internal::foldl` ?
Same holds for the other operations in this block.

Reply:
Yes, at the expense of using one more Scalar< D >, and unpacking it at the end.